src/plonk/prover.rs

```diff
@@ -330,13 +331,96 @@ pub fn create_proof<
         })
         .collect::<Result<Vec<_>, _>>()?;
 
+    // Create polynomial evaluator context for values.
+    let mut value_evaluator = poly::new_evaluator(|| {});
+
+    // Register fixed values with the polynomial evaluator.
+    let fixed_values: Vec<_> = pk
+        .fixed_values
+        .iter()
+        .map(|poly| value_evaluator.register_poly(poly.clone()))
+        .collect();
+
+    // Register advice values with the polynomial evaluator.
+    let advice_values: Vec<_> = advice
+        .iter()
+        .map(|advice| {
+            advice
+                .advice_values
+                .iter()
+                .map(|poly| value_evaluator.register_poly(poly.clone()))
```

Comment:
I think it should be possible for us to avoid these clones if we're able to consume these values, since I believe we don't need these after this registration has taken place. But this can happen separately.